src/CommandingOfficers/CommanderAbility.java

```diff
@@ -60,21 +68,59 @@ public final void activate(MapMaster gameMap)
       System.out.println("WARNING!: Performing ability with insufficient ability power!");
     }
 
-    myCommander.activateAbility(this);
+    myCommander.activateAbility(this, gameMap);
+    applyUnitModifiers(myCommander, gameMap);
 
     adjustCost();
     perform(gameMap);
   }
+  private final void applyUnitModifiers(Commander co, MapMaster gameMap)
+  {
+    if(!modsApplied.containsKey(co)) {
+      ArrayList<UnitModifier> unitModsToApply = new ArrayList<>();
+      enqueueUnitMods(gameMap, unitModsToApply);
+      modsApplied.put(co, unitModsToApply);
+      for(UnitModifier mod : unitModsToApply)
+        co.addUnitModifier(mod);
+    }
+  }
 
-  /** Subclasses will override this method to enact the ability's effects.
-   * Note that `getEvents` will be called before `perform`, so events can be cached. */
-  protected abstract void perform(MapMaster gameMap);
+  /**
+   * Public hook to handle cleanup.
+   */
+  public final void deactivate(MapMaster gameMap)
+  {
+    for( Commander co : new ArrayList<Commander>(modsApplied.keySet()) )
+    {
+      ArrayList<UnitModifier> modsToDrop = modsApplied.remove(co);
+      // Revert in reverse order, just to be safe
+      for( int i = modsToDrop.size() - 1; i >= 0; --i )
+        co.removeUnitModifier(modsToDrop.get(i));
+    }
+    revert(gameMap);
+  }
+
+  /** Subclasses will override this method to enact any ability effects that don't go in an event or a UnitModifier.
+   * <p>Note: getEvents() will be called before perform()*/
+  // Defaults to nothing since most powers don't need to directly affect game state
+  protected void perform(MapMaster gameMap)
+  {}
+  // Defaults to nothing since most powers don't need special revert logic.
+  protected void revert(MapMaster gameMap)
+  {}
 
   /** Allows a CommanderAbility to generate events that will be animated and published. */
   public GameEventQueue getEvents(MapMaster gameMap)
   {
     return new GameEventQueue();
   }
+  public GameEventQueue getRevertEvents(Commander co, MapMaster gameMap)
```

Comment:
Is this needed? It's never overridden or called anywhere

Reply:
Good catch. I do want to support the possibility for future things, though I guess it's not strictly necessary.
The usecase that springs to mind at the moment is "Iron Will's new HP-loss-on-revert effect could be done via events and actually animate that way"

Reply:
If we want to have the hook in place, we should be calling it wherever it's supposed to be invoked.

Though this is the sort of thing I probably wouldn't add at all until I had at least one active client.

Reply:
Done.